samples/sample-plugin-json/che-sample-plugin-json-ide/src/main/java/org/eclipse/che/plugin/jsonexample/ide/editor/JsonExampleCodeAssistProcessor.java

```diff
@@ -0,0 +1,85 @@
+ *******************************************************************************/
+package org.eclipse.che.plugin.jsonexample.ide.editor;
+
+
+import com.google.inject.Inject;
+
+import org.eclipse.che.ide.api.editor.codeassist.CodeAssistCallback;
+import org.eclipse.che.ide.api.editor.codeassist.CodeAssistProcessor;
+import org.eclipse.che.ide.api.editor.codeassist.CompletionProposal;
+import org.eclipse.che.ide.api.editor.texteditor.TextEditor;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.Unmarshallable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * JSON Example specific code assist processor.
+ */
+public class JsonExampleCodeAssistProcessor implements CodeAssistProcessor {
+
+    private final JsonExampleCodeAssistClient  client;
+    private final Unmarshallable<List<String>> unmarshaller;
+    private final NotificationManager          notificationManager;
+
+    /**
+     * Constructor.
+     *
+     * @param client
+     *         the client for retrieving completions
+     * @param notificationManager
+     *         the notification manager that is used for displaying
+     *         any errors in case computing completions fails
+     */
+    @Inject
+    public JsonExampleCodeAssistProcessor(final JsonExampleCodeAssistClient client,
+                                          final NotificationManager notificationManager
+                                         ) {
+        this.client = client;
+        this.unmarshaller = new StringListUnmarshaller();
+        this.notificationManager = notificationManager;
+    }
+
+    @Override
+    public void computeCompletionProposals(final TextEditor editor, final int offset, final CodeAssistCallback callback) {
+        final List<CompletionProposal> proposals = new ArrayList<>();
+
+        proposals.addAll(Arrays.asList(
+                new SimpleCompletionProposal("firstName"),
+                new SimpleCompletionProposal("lastName"),
+                new SimpleCompletionProposal("age")));
+
+        client.computeProposals(
+                new AsyncRequestCallback<List<String>>(unmarshaller) {
+                    @Override
+                    protected void onSuccess(List<String> additionalProposals) {
+                        for (String additionalProposal : additionalProposals) {
+                            proposals.add(new SimpleCompletionProposal(additionalProposal));
+                        }
+                        callback.proposalComputed(proposals);
+                    }
+
+                    @Override
+                    protected void onFailure(Throwable exception) {
+                        notificationManager.notify(exception.getMessage());
+                    }
+                });
+    }
+
+    @Override
+    public String getErrorMessage() {
+        return null;
```

Comment:
I guess instead of using the notificationManager we could also assign the exception message to an `errorMessage` field which is then returned in `getErrorMessage` and cleared in case the completion completed successfully. Would that make sense?